src/test/scala/mesosphere/marathon/core/externalvolume/impl/providers/DVDIProviderRootGroupValidationTest.scala

```diff
@@ -0,0 +1,133 @@
+package mesosphere.marathon.core.externalvolume.impl.providers
+
+import com.wix.accord.{ RuleViolation, Success, Failure, Result }
+import mesosphere.marathon.api.v2.Validation
+import mesosphere.marathon.core.externalvolume.ExternalVolumes
+import mesosphere.marathon.state._
+import org.apache.mesos.{ Protos => MesosProtos }
+import org.scalatest.{ FunSuite, GivenWhenThen, Matchers }
+import play.api.libs.json.{ JsString, Json }
+
+import scala.collection.immutable.Seq
+
+class DVDIProviderRootGroupValidationTest extends FunSuite with Matchers with GivenWhenThen {
+
+  test("two volumes with different names should not result in an error") {
+    val f = new Fixture
+    Given("a root group with two apps and conflicting volumes")
+    val rootGroup = Group(
+      id = PathId.empty,
+      groups = Set(
+        Group(
+          id = PathId("/nested"),
+          apps = Set(
+            f.appWithDVDIVolume(appId = PathId("/nested/foo1"), volumeName = "vol1"),
+            f.appWithDVDIVolume(appId = PathId("/nested/foo2"), volumeName = "vol2")
+          )
+        )
+      )
+    )
+
+    f.checkResult(
+      rootGroup,
+      expectedViolations = Set.empty
+    )
+  }
+
+  test("two volumes with same name result in an error") {
+    val f = new Fixture
+    Given("a root group with two apps and conflicting volumes")
+    val app1 = f.appWithDVDIVolume(appId = PathId("/nested/app1"), volumeName = "vol")
+    val app2 = f.appWithDVDIVolume(appId = PathId("/nested/app2"), volumeName = "vol")
+    val rootGroup = Group(
+      id = PathId.empty,
+      groups = Set(
+        Group(
+          id = PathId("/nested"),
+          apps = Set(app1, app2)
+        )
+      )
+    )
+
+    f.checkResult(
+      rootGroup,
+      expectedViolations = Set(
+        RuleViolation(
+          value = app1.externalVolumes.head,
+          constraint = "Volume name 'vol' in /nested/app1 conflicts with volume(s) of same name in app(s): /nested/app2",
+          description = Some("/groups(0)/apps(0)/externalVolumes(0)")
+        ),
+        RuleViolation(
+          value = app2.externalVolumes.head,
+          constraint = "Volume name 'vol' in /nested/app2 conflicts with volume(s) of same name in app(s): /nested/app1",
+          description = Some("/groups(0)/apps(1)/externalVolumes(0)")
+        )
+      )
+    )
+  }
+
+  class Fixture {
```

Comment:
We use a similar pattern elsewhere. If your `Fixture` contains state, it is nice to use a new instance for every test run.